src/SQLite.cs

```diff
@@ -1946,77 +1941,440 @@ protected internal void Dispose()
+		{
+			for (int i = 0; i < properties.Length; i++)
+			{
+				Index(properties[i], false, null, i);
+			}
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> Index(string indexName, params Expression<Func<TEntity, object>>[] properties)
+		{
+			for (int i = 0; i < properties.Length; i++)
+			{
+				Index(properties[i], false, indexName, i);
+			}
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> Unique(params Expression<Func<TEntity, object>>[] properties)
+		{
+			for (int i = 0; i < properties.Length; i++)
+			{
+				Index(properties[i], true, null, i);
+			}
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> Unique(string indexName, params Expression<Func<TEntity, object>>[] properties)
+		{
+			for (int i = 0; i < properties.Length; i++)
+			{
+				Index(properties[i], true, indexName, i);
+			}
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> PrimaryKey(Expression<Func<TEntity, object>> property, bool autoIncrement = false)
+		{
+			_primaryKey = property.AsPropertyInfo();
+			if (autoIncrement)
+			{
+				_autoInc.Add(_primaryKey);
+			}
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> Ignore(Expression<Func<TEntity, object>> property)
+		{
+			_ignore.AddProperty(property);
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> Ignore(params Expression<Func<TEntity, object>>[] properties)
+		{
+			_ignore.AddProperties(properties);
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> AutoIncrement(Expression<Func<TEntity, object>> property)
+		{
+			_autoInc.AddProperty(property);
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> AutoIncrement(params Expression<Func<TEntity, object>>[] properties)
+		{
+			_autoInc.AddProperties(properties);
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> NotNull(Expression<Func<TEntity, object>> property)
+		{
+			_notNull.AddProperty(property);
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> NotNull(params Expression<Func<TEntity, object>>[] properties)
+		{
+			_notNull.AddProperties(properties);
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> StoreAsText(Expression<Func<TEntity, object>> property)
+		{
+			_storeAsText.AddProperty(property);
+			return this;
+		}
+
+		public TableMappingBuilder<TEntity> StoreAsText(params Expression<Func<TEntity, object>>[] properties)
+		{
+			_storeAsText.AddProperties(properties);
+			return this;
+		}
+
+		/// <summary>
+		/// Executes a "create table if not exists" on the database. It also
+		/// creates any specified indexes on the columns of the table. It uses
+		/// a schema automatically generated from the specified type. You can
+		/// later access this schema by calling GetMapping.
+		/// </summary>
+		/// <returns>
+		/// The number of entries added to the database schema.
+		/// </returns>
+		public int CreateTable(CreateFlags createFlags = CreateFlags.None)
```

Comment:
I want to remove this and instead introduce a `CreateTable(TableMapping map)` method that registers the mapping if needed but then hands off to the normal code.